codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -782,11 +783,49 @@ picToObj' pic =
             return $ (unsafeCoerce arr :: JSVal)
     setProps xs obj = liftIO $ void $ mapM (\(s, v) -> setProp s v obj) xs
 
+trim :: Int -> String -> String
```

Comment:
Nitpicking here, but:

* You don't need the outer parentheses when defining `mid`
* You don't need parentheses around the expression between `case` and `of`
* You don't need the outer parentheses on either branch of the `case` (lines 789 and 790).